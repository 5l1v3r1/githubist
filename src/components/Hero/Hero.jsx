// @flow

import React from 'react';

import styles from './Hero.scss';

import { Container } from '..';

import githubLogo from '../../img/github.svg';

type Props = {
  description: string,
};

const Hero = ({ description }: Props) => (
  <div className={styles.hero}>
    <Container>
      <div className={styles.inner}>
        <div>
          <h6 className={styles.badge}>2018</h6>
          <h1 className={styles.title}>
            Github
            <br />
            Türkiye İstatistikleri
          </h1>
          <p className={styles.description}>{description}</p>
        </div>

        <img src={githubLogo} className={styles.image} alt="Github" width={431} height={420} />
      </div>
    </Container>
  </div>
);

export default Hero;
