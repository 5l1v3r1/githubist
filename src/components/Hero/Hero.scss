@import '../../theme/theme';

.hero {
  padding: $section-gap * 2 0;
}

.inner {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.badge {
  margin: 0 0 map-get($spacers, 2x-small) 0;
  color: $primary-color;
  font-size: map-get($heading-sizes, h6);
}

.title {
  margin: 0 0 map-get($spacers, x-small) 0;
  color: $black-color;
  font-size: map-get($heading-sizes, h1);
  font-weight: 600;
  line-height: $heading-line-height;
}

.image {
  margin-left: map-get($spacers, 2x-large);
}
