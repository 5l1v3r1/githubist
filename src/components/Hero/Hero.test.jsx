// @flow

import React from 'react';
import { shallow } from 'enzyme';

import Hero from './Hero';

it('renders correctly', () => {
  const wrapper = shallow(<Hero description="Lorem ipsum dolar sit amet." />);

  expect(wrapper).toMatchSnapshot();
});
