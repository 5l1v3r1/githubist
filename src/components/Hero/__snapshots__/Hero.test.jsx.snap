// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <Container
    className=""
  >
    <div>
      <div>
        <h6>
          2018
        </h6>
        <h1>
          Github
          <br />
          Türkiye İstatistikleri
        </h1>
        <p>
          Lorem ipsum dolar sit amet.
        </p>
      </div>
      <img
        alt="Github"
        height={420}
        src={
          Object {
            "process": [Function],
          }
        }
        width={431}
      />
    </div>
  </Container>
</div>
`;
